lib/request.js

```diff
@@ -461,6 +461,51 @@ defineGetter(req, 'stale', function stale(){
 });
 
 /**
+ * Check if request is local, aka
+ * if it originates from localhost
+ *
+ * @reutrn {Boolean}
+ * @public
+ */
+
+defineGetter(req, 'local', function local(){
+  var ip = this.ip;
+
+  return [ '127.0.0.1', '::', '::1', 'fe80::1' ].some(function (a) { return a === ip; });
```

Comment:
This will fail for non-normalized values, like `FE80::1` or even `fe80:0:0:0:0:0:0:1`; it also doesn't detect the IPv4-mapped IPV6 addresses Node.js 0.12+ provided on dual stacks, which is `::ffff:127.0.0.1`